bukuserver/templates/bukuserver/lib.html

```diff
@@ -58,3 +58,58 @@
   }</script>
   {% endif %}
 {% endmacro %}
+
+{% macro set_lang() %}
+  <script>document.documentElement.lang = {{ lang|tojson }}</script>
+{% endmacro %}
+
+{% macro fix_translations(list) %}  {# invoke in *_list.html BEFORE tail.super()! #}
+  {{ set_lang() }}
+  {% set LISTS = {'tags': {'name': _p('tags', 'name'), 'usage_count': _p('tags', 'usage count')},
+                  'bookmarks': {'id': _p('bookmarks', 'index'), 'url': _p('bookmarks', 'url'), 'title': _p('bookmarks', 'title'),
+                                'tags': _p('bookmarks', 'tags'), 'order': _p('bookmarks', 'order')}} %}
+  <script>{
+    $(`button[title="Delete record"]`).attr('title', {{ _('Delete record')|tojson }});  // see flask-admin issue #1974
+    const FILTERS = {{ LISTS.get(list, {})|tojson }};
+    let watcher = new MutationObserver(xs => xs.forEach(x => x.addedNodes.forEach(e => {
+      if (!['filter-groups-data', 'active-filters-data'].includes(e.id)) return;
+      let data = JSON.parse(e.innerHTML);
+      let converted = (e.id == 'active-filters-data' ? data.map(([id, name, value]) => [id, FILTERS[name]||name, value]) :
+                       Object.fromEntries( Object.entries(data).map(([k, v]) => [FILTERS[k]||k, v]) ));
+      e.innerHTML = JSON.stringify(converted);
+    })));
+    watcher.observe(document.body, {childList: true, subtree: true});
+    setTimeout(() => watcher.disconnect());  // will stop observing once <body> is rendered
+    $(document).ready(() => {
+      $(`.field-filters .filter`).each(function () {
+        let text = $(this).text();
+        $(this).text(FILTERS[text] || text);
+      });
+    })
```

Comment:
Filter data is generated and printed out as raw JSON in Flask-Admin, then it's immediately used to initialize filters form. (And the add-filter dropdown is rendered separately.) More importantly, it _has to_ be translated on the frontend, because backend supports translations for filters but not filter groups.

Thus, to translate the filters data I had to substitute values in the JSON _immeadiately_ after they got rendered on page.